project-codeflare/codeflare-sdk
# Add UI table with Cluster specs and action buttons

# Issue link


Jira: https://issues.redhat.com/browse/RHOAIENG-12526 

# What changes have been made

![image](https://github.com/user-attachments/assets/8e740820-1de7-4487-b433-207cb4b6589a)
- Added UI table that will display existing clusters with their specs, and handle user interactions which include:
  - Selecting a cluster.
  - Deleting the selected cluster.
  - Opening in a new tab the Ray Dashboard of the selected cluster.
  - Opening in a new tab the Ray Dashboard Job view of the selected cluster.
- Formatted table to merge req~lim resources into one column.
- Formatted status field to change text color and add an icon based on the cluster status.
- Added unit tests.
- Added outputs.
- Updated Ray model with `cpu requests` and `cpu limits`.

TODO: 
- Add `view_clusters` to guided-notebooks by default? - Given `Cluster Up` and `Cluster Down` widgets will appear by default, I think it's a good idea to immediately encourage the user to use (and be aware) of this feature. We don't need to trigger the buttons for `test-guided-notebooks` tests as it will be covered by `ui-notebooks-test` tests. Only the expected outputs would have to be updated.
- Enhance unit tests.
- Add to UI tests
- Implement `TODO` left in `widgets.py` file. [ :heavy_check_mark: ]

# Verification steps



Install the codeflare-sdk and import:
```
from codeflare_sdk import view_clusters
```
There are several items to test which include:
- Create several clusters.
- Select a cluster.
- Click on each button.
- Switch clusters and click on each button.
- Verify spec displayed.
- Verify output.
- Repeat with AppWrappers.


## Checks
- Testing Strategy
